controllers/posts_controller.js

```diff
@@ -1,66 +1,60 @@
-const post = require('../model/posts_model')
-const utils = require('../utils/util')
+const post = require('../model/posts_model');
+const utils = require('../utils/util');
 
-module.exports.addPosts = ({body: {url = null, description = null, tags = null}, user = null}, res) => {
-    let errorMessage = '';
+module.exports.addPosts = async ({body: {url = null, description = null, tags = null}, user = null}, res) => {
+    let errors = [];
 
     if (utils.isEmpty(user.id)) {
-        errorMessage += 'User ID is Null\n'
+        errors.push('User ID is Null');
     }
     if (utils.isEmpty(url)) {
-        errorMessage += 'URL is Null\n'
+        errors.push('URL is Null');
     }
 
-    if (!utils.isEmpty(errorMessage)) {
-        res.status(401).json({
+    if (errors.length > 0) {
+        res.status(400).json({
             error: true,
-            data: errorMessage
+            data: errors.join('\n')
         })
     } else {
-        const addPostObservable = post.addPosts(description, url, tags, user.id);
-        addPostObservable.subscribe(
-            post =>
-                res.status(200).json({
-                    error: false,
-                    data: post
-                }),
-            err => {
-                console.error(err);
-                res.status(400).json({
-                    error: true,
-                    data: null
-                })
-            }
-        );
-    }
-}
-
-module.exports.getPostByID = ({params: id = null}, res) => {
-    const getPostObservable = post.getPostByID(id);
-    getPostObservable.subscribe(
-        next => {
-            let [post, user] = next
-            post.user_info = user
+        try {
+            const post = await post.addPosts(description, url, tags, user.id);
```

Comment:
`post = await post.addPosts()` .......

Reply:
Yeah I found that but didn't push the changes yet

Reply:
Yeah, I found that but haven't pushed the changes yet